src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/BlobListenerFactory.cs

```diff
@@ -198,50 +197,103 @@ internal class BlobListenerFactory : IListenerFactory
                 // want a single instance of the blob poll/scan logic to be running
                 // across host instances
-                var singletonBlobListener = _singletonManager.CreateHostSingletonListener(
-                    new BlobListener(sharedBlobListener), SingletonBlobListenerScopeId);
+                var singletonBlobListener = _singletonManager.CreateHostSingletonListener(sharedBlobListener, SingletonBlobListenerScopeId);
                 _sharedContextProvider.SetValue(SingletonBlobListenerScopeId, true);
-
-                return new CompositeListener(singletonBlobListener, queueListener);
+                // since now that queuelistener is null, we no longer need CompositeListener
+                return singletonBlobListener;
             }
             else
             {
                 // We've already created the singleton blob listener
                 // so just return our queue listener. Note that while we want the
                 // blob scan to be singleton, the shared queue listener needs to run
                 // on ALL instances so load can be scaled out
-                return queueListener;
+                return new NullListener();
             }
         }
 
         private async Task RegisterWithSharedBlobListenerAsync(
             string hostId,
             SharedBlobListener sharedBlobListener,
             IStorageBlobClient blobClient,
-            IStorageQueue hostBlobTriggerQueue,
-            IMessageEnqueuedWatcher messageEnqueuedWatcher,
+            DispatchProcessor dispatchProcessor,
             CancellationToken cancellationToken)
         {
             BlobTriggerExecutor triggerExecutor = new BlobTriggerExecutor(hostId, _functionId, _input,
-                BlobETagReader.Instance, new BlobReceiptManager(blobClient),
-                new BlobTriggerQueueWriter(hostBlobTriggerQueue, messageEnqueuedWatcher));
+                BlobETagReader.Instance, new BlobReceiptManager(blobClient), dispatchProcessor);
 
             await sharedBlobListener.RegisterAsync(_container, triggerExecutor, cancellationToken);
         }
 
-        private void RegisterWithSharedBlobQueueListenerAsync(
-            SharedBlobQueueListener sharedBlobQueueListener,
-            IStorageBlobClient blobClient,
-            IStorageQueueClient queueClient)
+        private class BlobMessageHandler : MessageHandler
         {
-            BlobQueueRegistration registration = new BlobQueueRegistration
+            private readonly IStorageBlobClient _blobClient;
+            private readonly ITriggeredFunctionExecutor _executor;
+            private readonly IBlobWrittenWatcher _blobWrittenWatcher;
+            internal BlobMessageHandler(ITriggeredFunctionExecutor executor,
+                                        IStorageBlobClient blobClient,
+                                        IStorageQueueClient queueClient,
+                                        IBlobWrittenWatcher blobWrittenWatcher)
```

Comment:
I don't understand why your message handler is taking this watcher. The semantics before were that the blob scanner would use this to notify when it discovered a new blob. I don't think we need to change that.